pkg/services/cloudmigration/api/api.go

```diff
@@ -343,3 +359,186 @@ func (cma *CloudMigrationAPI) DeleteSession(c *contextmodel.ReqContext) response
+	}
+
+	ss, err := cma.cloudMigrationService.CreateSnapshot(ctx, uid)
+	if err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "error creating snapshot", err)
+	}
+
+	return response.JSON(http.StatusOK, CreateSnapshotResponseDTO{
+		SnapshotUID: ss.UID,
+	})
+}
+
+// swagger:route GET /cloudmigration/migration/{uid}/snapshot/{snapshotUid} migrations getSnapshot
+//
+// Get metadata about a snapshot, including where it is in its processing and final results.
+//
+// Responses:
+// 200: getSnapshotResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 500: internalServerError
+func (cma *CloudMigrationAPI) GetSnapshot(c *contextmodel.ReqContext) response.Response {
+	ctx, span := cma.tracer.Start(c.Req.Context(), "MigrationAPI.GetSnapshot")
+	defer span.End()
+
+	sessUid, snapshotUid := web.Params(c.Req)[":uid"], web.Params(c.Req)[":snapshotUid"]
+	if err := util.ValidateUID(sessUid); err != nil {
+		return response.ErrOrFallback(http.StatusBadRequest, "invalid session uid", err)
+	}
+	if err := util.ValidateUID(snapshotUid); err != nil {
+		return response.ErrOrFallback(http.StatusBadRequest, "invalid snapshot uid", err)
+	}
+
+	snapshot, err := cma.cloudMigrationService.GetSnapshot(ctx, sessUid, snapshotUid)
+	if err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "error retrieving snapshot", err)
+	}
+
+	result, err := snapshot.GetSnapshotResult()
+	if err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "error snapshot reading snapshot results", err)
+	}
+
+	dtoResults := make([]MigrateDataResponseItemDTO, len(result))
+	for i := 0; i < len(result); i++ {
+		dtoResults[i] = MigrateDataResponseItemDTO{
+			Type:   MigrateDataType(result[i].Type),
+			RefID:  result[i].RefID,
+			Status: ItemStatus(result[i].Status),
+			Error:  result[i].Error,
+		}
+	}
+
+	respDto := GetSnapshotResponseDTO{
+		SnapshotDTO: SnapshotDTO{
+			SnapshotUID: snapshot.UID,
+			Status:      fromSnapshotStatus(snapshot.Status),
+			SessionUID:  sessUid,
+			Created:     snapshot.Created,
+			Finished:    snapshot.Finished,
+		},
+		Results: dtoResults,
+	}
+
+	return response.JSON(http.StatusOK, respDto)
+}
+
+// swagger:route GET /cloudmigration/migration/{uid}/snapshot migrations getShapshotList
+//
+// Get a list of snapshots for a session.
+//
+// Responses:
+// 200: snapshotListResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 500: internalServerError
+func (cma *CloudMigrationAPI) GetSnapshotList(c *contextmodel.ReqContext) response.Response {
+	ctx, span := cma.tracer.Start(c.Req.Context(), "MigrationAPI.GetShapshotList")
+	defer span.End()
+
+	uid := web.Params(c.Req)[":uid"]
+	if err := util.ValidateUID(uid); err != nil {
+		return response.ErrOrFallback(http.StatusBadRequest, "invalid session uid", err)
+	}
+
+	snapshotList, err := cma.cloudMigrationService.GetSnapshotList(ctx, uid)
```

Comment:
Probably good to paginate endpoints that can return an unbounded number of items. Basic OFFSET + LIMIT pagination should be enough

Reply:
cool thanks for the suggestion! done